app/scripts/components/analysis/define/use-stac-collection-search.ts

```diff
@@ -37,10 +33,31 @@ export function useStacCollectionSearch({
   const result = useQuery({
     queryKey: ['stacCollection'],
     queryFn: async ({ signal }) => {
-      const collectionResponse = await axios.get(collectionUrl, {
-        signal
-      });
-      return collectionResponse.data.collections;
+      const collectionUrlsFromDataSets = allAvailableDatasetsLayers
+        .filter((dataset) => dataset.stacApiEndpoint)
+        .map((dataset) => `${dataset.stacApiEndpoint}/collections`)
+        .filter((value, index, array) => array.indexOf(value) === index);
+
```

Comment:
We can probably use the `reduce` method instead to simplify the first two iterations, not sure what the point of the last one is

```
const collectionUrlsFromDataSets = allAvailableDatasetsLayers.reduce((filtered, dataset) => {
   if (dataset.stacApiEndpoint) {
      filtered.push(`${dataset.stacApiEndpoint}/collections`)
   }
return filtered;
}, [])
```

Reply:
I like `filter` + `map` when there is no shared condition between them. (I think it makes the code more straightforward to read. lmk if you think differently though!) But hm it seems the last condition is not clear. It is there to pick only unique values. We need this condition because multiple layers can point the same stac api endpoint. I tried a different approach with `Set` which I am not really sure is easy to read, but probably at least make it exclusive about what it is trying to do? lmk if you have a better idea!

Reply:
I also have a preference for `reduce`. IMO it is not harder to read and avoids array cycles. Fair to say that in this case it has no impact, but the principle stands.
However I'd avoid mutating methods:
```ts
const collectionUrlsFromDataSets = allAvailableDatasetsLayers.reduce((filtered, { stacApiEndpoint }) => {
  returnstacApiEndpoint ? [...filtered, `${stacApiEndpoint}/collections`] : filtered;
}, [])
```

Reply:
oki, changed this prt to use `reduce`. Thanks for your inputs!